### PR TITLE
PROD-6214 : Profile Avatar User Initials 

### PR DESCRIPTION
**Pull request by milanhirapra**
### Jira Issue:

https://buddyboss.atlassian.net/browse/PROD-6214
https://buddyboss.atlassian.net/browse/PROD-6366

### General Note
Keep all conversations related to this PR in the associated Jira issue(s). Do NOT add comment on this PR or edit this PR’s description.

### Notes to Developer
* Ensure the IDs (i.e. PROD-1) of all associated Jira issues are reference in this PR’s title
* Ensure that you have achieved the Definition of Done before submitting for review
* When this PR is ready for review, move the associate Jira issue(s) to “Needs Review” (or “Code Review” for Dev Tasks)

### Notes to Reviewer
* Ensure that the Definition of Done have been achieved before approving a PR
* When this PR is approved, move the associated Jira issue(s) to “Needs QA” (or “Approved” for Dev Tasks)

### CODE REVIEW

File: src/class-buddypress.php [hunk 1]
```diff
@@ -330,7 +330,7 @@ private function setup_globals() {
 
 		/** Versions */
 		$this->version    = defined( 'BP_PLATFORM_VERSION' ) ? BP_PLATFORM_VERSION : ( defined( 'BP_VERSION' ) ? BP_VERSION : '1.0.0' );
-		$this->db_version = 20761;
+		$this->db_version = 20764;
```
Comment (KartikSuthar) [inline src/class-buddypress.php:333]: revert this.

File: src/bp-core/bp-core-update.php [hunk 2]
```diff
@@ -479,6 +479,10 @@ function bp_version_updater() {
 			bb_update_to_2_4_60();
 		}
 
+		if ( $raw_db_version < 20764 ) {
+			bb_update_to_2_4_70();
+		}
+
```
Comment (KartikSuthar) [inline src/bp-core/bp-core-update.php:485]: remove this.

File: src/bp-core/bp-core-update.php [hunk 3]
```diff
@@ -3401,3 +3405,19 @@ function bb_update_to_2_4_60() {
...
+ * @return void
+ */
+function bb_update_to_2_4_70() {
+	wp_cache_flush();
+	// Purge all the cache for API.
+	if ( class_exists( 'BuddyBoss\Performance\Cache' ) ) {
+		// Clear members API cache.
+		BuddyBoss\Performance\Cache::instance()->purge_by_component( 'bp-members' );
+	}
+}
```
Comment (KartikSuthar) [inline src/bp-core/bp-core-update.php:3423]: remove this, we don't want to clear the API cache on update.
We just need to clear the cache when user changes the settings for "Default Profile Avatar" or "Default Group Avatar"
https://prnt.sc/jYPYMqk0uAkX

File: src/bp-core/bp-core-functions.php [hunk 4]
```diff
@@ -8993,3 +8999,438 @@ function bb_load_reaction() {
...
+		$file_url  = bp_core_get_upload_dir( 'url' ) . '/avatars/default/' . $r['item_id'] . '/' . $filename;
+	}
+
+	if ( ! is_object( $wp_filesystem ) ) {
+		if ( ! function_exists( 'WP_Filesystem' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+		}
+
+		WP_Filesystem();
+	}
```
Comment (KartikSuthar) [inline src/bp-core/bp-core-functions.php:9305]: please create a function for this which can return the object $wp_filesystem and use whenever necessary.

**Review by KartikSuthar — CHANGES_REQUESTED** [review]

**Review by KartikSuthar — DISMISSED** [review]

**Review by KartikSuthar — APPROVED** [review]

**Review by jitendrabanjara1991 — DISMISSED** [review]

**Pull request merged**